src/Application.vala

```diff
@@ -46,6 +46,15 @@ public class Camera.Application : Gtk.Application {
     protected override void activate () {
         if (this.get_windows () == null) {
             main_window = new MainWindow (this);
+
+            var window_size = settings.get_value ("window-size");
```

Comment:
Instead of having this stuff in between two references to `main_window` it should be all above it so that all the references to `main_window` are grouped like:

```
[all the rect stuff]

main_window = new MainWindow (this);
main_window.set_allocation (rect);
main_window.show_all ();
```